cpp/include/cudf/io/parquet.hpp

```diff
@@ -988,18 +1024,23 @@ class parquet_chunked_writer {
    * @brief Writes table to output.
    *
    * @param[in] table Table that needs to be written
+   * @param[in] partitions Optional partitions to divide the table into. If specified, must be same
+   * size as number of sinks.
    * @return returns reference of the class object
    */
-  parquet_chunked_writer& write(table_view const& table);
+  parquet_chunked_writer& write(
+    table_view const& table, std::vector<std::pair<size_type, size_type>> const& partitions = {});
 
   /**
    * @brief Finishes the chunked/streamed write process.
    *
-   * @param[in] column_chunks_file_path Column chunks file path to be set in the raw output metadata
+   * @param[in] column_chunks_file_paths Column chunks file path to be set in the raw output
+   * metadata
    * @return A parquet-compatible blob that contains the data for all rowgroups in the list only if
-   * `column_chunks_file_path` is provided, else null.
+   * `column_chunks_file_paths` is provided, else null.
    */
-  std::unique_ptr<std::vector<uint8_t>> close(std::string const& column_chunks_file_path = "");
+  std::unique_ptr<std::vector<uint8_t>> close(
```

Comment:
Should we make this `std::optional`? I know it's unrelated, but since we're touching this API, might as well break it some more.

Reply:
Then the caller would have to pass `std::nullopt`. Is this desired?

Reply:
I was referring to the return type, now I see that the input parameter has similar semantics :)
My understanding is that we're using `unique_ptr` here just to be able to return null. `optional` is a better tool for that.

I agree, `column_chunks_file_paths` is fine as-is. Having a default value makes the use very clean with and without paths.

Reply:
I added this in https://github.com/rapidsai/cudf/pull/9810/commits/d0de9a9dbd7efdb70ad9d7d86fc67b55e63fb9e5 but had to revert in https://github.com/rapidsai/cudf/pull/9810/commits/5a17a4cfc0cca243e2fb2ddc65439303523572aa because cython doesn't support `std::optional` yet.